client/src/components/sources/sourceListItem.js

```diff
@@ -52,6 +65,31 @@ class SourceListItem extends React.Component {
     });
   }
 
+  loadExpandData(expandType) {
+    const { item } = this.props;
+
+    if (expandType === 'okHosts' || expandType === 'failedHosts') {
+      this.setState({
+        scanResultsPending: true,
+        scanResultsError: null
+      });
+      this.props
+        .getScanResults(item.connection.id)
+        .then(results => {
+          this.setState({
+            scanResultsPending: false,
+            scanResults: _.get(results.value, 'data')
+          });
+        })
+        .catch(error => {
+          this.setState({
+            scanResultsPending: false,
+            scanResultsError: helpers.getErrorMessageFromResults(error.payload)
+          });
+        });
+    }
+  }
+
```

Comment:
Note: this should tie in nicely with, hopefully, some of the last bits of the wizard and the inspect scan